providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/steps/StepDefinitions.java

```diff
@@ -374,4 +388,107 @@ public void the_returned_value_should_be(String expected) {
         String value = client.getStringValue(this.stringFlagKey, this.stringFlagDefaultValue, this.customEvaluatorContext);
         assertEquals(expected, value);
     }
+
+    /*
+     * Events
+     */
+
+     // Flag change event
+    @When("a PROVIDER_CONFIGURATION_CHANGED handler is added")
+    public void a_provider_configuration_changed_handler_is_added() {
+        this.changeHandler = (EventDetails details) -> { 
+            this.isChangeHandlerRun = true;
+        };
+        client.onProviderConfigurationChanged(this.changeHandler);
+
+    }
+    @When("a flag with key {string} is modified")
+    public void a_flag_with_key_is_modified(String flagKey) {
+        //This happens automatically
+    }
+
+    @Then("the PROVIDER_CONFIGURATION_CHANGED handler must run")
+    public void the_provider_configuration_changed_handler_must_run() {
+        Awaitility.await()
+        .atMost(Duration.ofSeconds(2))
+        .until(() -> {
+            return this.isChangeHandlerRun;
+        });
+    }
+
+    @Then("the event details must indicate {string} was altered")
+    public void the_event_details_must_indicate_was_altered(String flagKey) {
+        //TODO: In-process-provider doesnt support flag change list.
+    }
+
+    // Provider ready event
+    @When("a PROVIDER_READY handler is added")
+    public void a_provider_ready_handler_is_added() {
+        this.readyHandler = (EventDetails details) -> { 
+            this.isReadyHandlerRun = true;
+        };
+        client.onProviderReady(this.readyHandler);
+    }
+
+    @Then("the PROVIDER_READY handler must run")
+    public void the_provider_ready_handler_must_run() {
+        Awaitility.await()
+        .atMost(Duration.ofSeconds(2))
+        .until(() -> {
+            return this.isReadyHandlerRun;
+        });
+    }
+
+    /*
+    * Zero Value
+    */
+
+    // boolean value
+    @When("a zero-value boolean flag with key {string} is evaluated with default value {string}")
+    public void a_zero_value_boolean_flag_with_key_is_evaluated_with_default_value(String flagKey, String defaultValue) {
+        this.booleanFlagKey = flagKey;
+        this.booleanFlagDefaultValue = Boolean.valueOf(defaultValue);
+    }
+    @Then("the resolved boolean zero-value should be {string}")
+    public void the_resolved_boolean_zero_value_should_be(String expected) {
+        boolean value = client.getBooleanValue(this.booleanFlagKey, this.booleanFlagDefaultValue);
+         assertEquals(Boolean.valueOf(expected),value);   
+    }
+
+    // float/double value
+    @When("a zero-value float flag with key {string} is evaluated with default value {double}")
+    public void a_zero_value_float_flag_with_key_is_evaluated_with_default_value(String flagKey, Double defaultValue) {
+        // TODO: There is a bug here with 0 value floats  
```

Comment:
When testing locally defaultValue will not produce 0.0. Instead defaultValue is stuck at 0. Only the in-process evaluator seems to be impacted.